src/misc/setup.cpp

```diff
@@ -1024,6 +1042,24 @@ bool CommandLine::FindCommand(unsigned int which,std::string & value) {
 	return true;
 }
 
+// Was a directory provided on the command line?
+bool CommandLine::HasDirectory() const
+{
+	for (const auto& arg : cmds)
+		if (open_directory(arg.c_str()))
+			return true;
+	return false;
+}
+
+// Was an executable filename provided on the command line?
+bool CommandLine::HasExecutable() const
```

Comment:
This function will still return true for directories with executable extension. It's not a problem right now, because we want to treat them the same way as executable *in this feature*, but might be a problem for future users who will expect this function to return true only for executable files.

To not overblow the PR - maybe a simple function name change for now? `HasExecutableName`?

Reply:
Good catch; 
Yes, makes sense that this function's name indicate it only inspects names and not the file itself.  Superficial, and can be tricked - but hopefully fine in 99% of cases (and the consequence of it being wrong aren't harmful, just perhaps unexpected).